packages/php-wasm/universal/src/lib/load-php-runtime.ts

```diff
@@ -145,20 +143,24 @@ export async function loadPHPRuntime(
 			if (phpModuleArgs.onRuntimeInitialized) {
 				phpModuleArgs.onRuntimeInitialized();
 			}
-			resolvePhpReady();
+			resolvePHP();
 		},
 		monitorRunDependencies(nbLeft) {
 			if (nbLeft === 0) {
 				delete PHPRuntime.monitorRunDependencies;
-				resolveDepsReady();
+				resolveDeps();
 			}
 		},
 	});
-	for (const { default: loadDataModule } of dataDependenciesModules) {
-		loadDataModule(PHPRuntime);
-	}
+
+	await Promise.all(
+		dataDependenciesModules.map(({ default: dataModule }) =>
+			dataModule(PHPRuntime)
+		)
+	);
```

Comment:
before we have been running these and moving on possibly before they finish, but now with `Promise.all` we are pausing until they all finish. is that intentional?

also, we are using `Promise.all` so if a single data dependency throws an error the entire thing collapses.

 - do we want to use `Promise.allSettled` so we can report failures for specific dependencies?
 - if not, do we want to report the failure of _any_ data dependency? because we're not wrapping this in a `try`/`catch` or adding any `catch` handler to any of the promises.

Reply:
Before we've waited for data dependencies, too, in one of the next lines: `await depsReady;`. That line is still there, in this PR, but I don't think we need it anymore.

> do we want to use Promise.allSettled so we can report failures for specific dependencies?

We could, but that would amount to the same outcome – display a loading error. Or would you like to do something more granular with that information?

Reply:
No, we still need `await depsReady;` because it's resolved after the data dependencies have been both loaded **and** processed

Reply:
> We could, but that would amount to the same outcome – display a loading error. Or would you like to do something more granular with that information?

the point was that `depsReady` is all or nothing and `.allSettled` can tell us which dependencies failed. so if our goal is to expose more errors it could be beneficial to say something like "dependency {name} from {URL} failed to download" and be able to do that for each one that fails.

Reply:
Ah, I see what you mean! The information about the URL is included in the error message thrown by the Emscripten module.  The ability to display multiple errors would still be nice, we’d just need to handle an array of error messages up the call stack.

Worth noting: data dependencies are produced by file_packager.json and we only load a single data dependency now: wp.data.